File: firebase-sample/src/main/java/ru/surfstudio/android/firebase/sample/ui/base/configurator/ActivityConfigurator.java
package ru.surfstudio.android.firebase.sample.ui.base.configurator;

import ru.surfstudio.android.core.ui.configurator.BaseActivityConfigurator;
import ru.surfstudio.android.firebase.sample.app.App;
import ru.surfstudio.android.firebase.sample.app.dagger.AppComponent;
import ru.surfstudio.android.firebase.sample.ui.base.dagger.activity.ActivityComponent;
import ru.surfstudio.android.firebase.sample.ui.base.dagger.activity.DaggerActivityComponent;
import ru.surfstudio.android.sample.dagger.ui.base.dagger.activity.DefaultActivityModule;

/**
 * Базовый конфигуратор для активити
 */
public class ActivityConfigurator extends BaseActivityConfigurator<ActivityComponent, AppComponent> {

    @Override
    protected ActivityComponent createActivityComponent(AppComponent parentComponent) {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .defaultActivityModule(new DefaultActivityModule(getPersistentScope()))
                .build();
    }

    @Override
    protected AppComponent getParentComponent() {
        return ((App) getTargetActivity().getApplicationContext()).getAppComponent();
    }
}

File: navigation/sample-standard/src/main/java/ru/surfstudio/android/navigation/sample_standard/dagger/ui/configurator/ActivityConfigurator.kt
package ru.surfstudio.android.navigation.sample_standard.dagger.ui.configurator

import ru.surfstudio.android.core.ui.configurator.BaseActivityConfigurator
import ru.surfstudio.android.navigation.sample_standard.App
import ru.surfstudio.android.navigation.sample_standard.dagger.AppComponent
import ru.surfstudio.android.navigation.sample_standard.dagger.AppNavigationModule
import ru.surfstudio.android.navigation.sample_standard.dagger.ui.ActivityComponent
import ru.surfstudio.android.navigation.sample_standard.dagger.ui.ActivityNavigationModule
import ru.surfstudio.android.navigation.sample_standard.dagger.ui.DaggerActivityComponent
import ru.surfstudio.android.sample.dagger.ui.base.dagger.activity.DefaultActivityModule

class ActivityConfigurator : BaseActivityConfigurator<ActivityComponent, AppComponent>() {

    @Override
    override fun createActivityComponent(parentComponent: AppComponent): ActivityComponent {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .defaultActivityModule((DefaultActivityModule(persistentScope)))
                .activityNavigationModule(ActivityNavigationModule())
                .build()
    }

    @Override
    override fun getParentComponent(): AppComponent {
        val app = persistentScope.screenState.activity.applicationContext as App
        return app.appComponent
    }
}
